pkg/toolsets/kubevirt/vm/create/tool.go

```diff
@@ -0,0 +1,781 @@
+						},
+						"performance": {
+							Type:        "string",
+							Description: "Optional performance family hint for the VM instance type (e.g., 'u1' for general-purpose, 'o1' for overcommitted, 'c1' for compute-optimized, 'm1' for memory-optimized). Defaults to 'u1' (general-purpose) if not specified.",
+							Examples:    []interface{}{"general-purpose", "overcommitted", "compute-optimized", "memory-optimized"},
+						},
+					},
+					Required: []string{"namespace", "name"},
+				},
+				Annotations: api.ToolAnnotations{
+					Title:           "Virtual Machine: Create",
+					ReadOnlyHint:    ptr.To(true),
+					DestructiveHint: ptr.To(false),
+					IdempotentHint:  ptr.To(true),
+					OpenWorldHint:   ptr.To(false),
+				},
+			},
+			Handler: create,
+		},
+	}
+}
+
+type vmParams struct {
+	Namespace           string
+	Name                string
+	ContainerDisk       string
+	Instancetype        string
+	Preference          string
+	UseDataSource       bool
+	DataSourceName      string
+	DataSourceNamespace string
+}
+
+type DataSourceInfo struct {
+	Name                string
+	Namespace           string
+	Source              string
+	DefaultInstancetype string
+	DefaultPreference   string
+}
+
+type PreferenceInfo struct {
+	Name string
+}
+
+type InstancetypeInfo struct {
+	Name   string
+	Labels map[string]string
+}
+
+func create(params api.ToolHandlerParams) (*api.ToolCallResult, error) {
+	// Parse and validate input parameters
+	createParams, err := parseCreateParameters(params)
+	if err != nil {
+		return api.NewToolCallResult("", err), nil
+	}
+
+	// Search for available DataSources
+	dataSources, _ := searchDataSources(params, createParams.Workload)
+
+	// Match DataSource based on workload input
+	matchedDataSource := matchDataSource(dataSources, createParams.Workload)
+
+	// Resolve preference from DataSource defaults or cluster resources
+	preference := resolvePreference(params, createParams.Preference, matchedDataSource, createParams.Workload, createParams.Namespace)
+
+	// Resolve instancetype from DataSource defaults or size/performance hints
+	instancetype := resolveInstancetype(params, createParams, matchedDataSource)
+
+	// Build template parameters from resolved resources
+	templateParams := buildTemplateParams(createParams, matchedDataSource, instancetype, preference)
+
+	// Render the VM creation plan template
+	result, err := renderTemplate(templateParams)
+	if err != nil {
+		return api.NewToolCallResult("", err), nil
+	}
+
+	return api.NewToolCallResult(result, nil), nil
+}
+
+// createParameters holds parsed input parameters for VM creation
+type createParameters struct {
+	Namespace    string
+	Name         string
+	Workload     string
+	Instancetype string
+	Preference   string
+	Size         string
+	Performance  string
+}
+
+// parseCreateParameters parses and validates input parameters
+func parseCreateParameters(params api.ToolHandlerParams) (*createParameters, error) {
```

Comment:
I'd suggest naming it as `parseCreateVirtualMachineParams`. Even if it's a private function and not exposed.